### PR TITLE
Upgrade to Capacitor 6

### LINKED ISSUE #354: Compatibility with capacitor 6
Hi,
I get this message with Capacitor 6:
Your version of @codetrix-studio/capacitor-google-auth is not compatible with @capacitor/core 6.0.0

### LINKED ISSUE #356: Capacitor 6 support
Android SDK 34+ will be mandatory after August 2024. Do you have plans to upgrade this plugin?

### PR DESCRIPTION
**Pull request by KerimM**
closes #354 
closes #356

### CODE REVIEW

**Review by gasci — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@codetrix-studio/capacitor-google-auth",
-  "version": "3.4.0-rc.0",
+  "version": "3.5.0-rc.0",
```
Comment (reslear) [inline package.json:3]: thi is already cap6 verision, please change to `3.4.0-rc.1`

**Comment by reslear** [comment]
I'll have to remember to return https://github.com/CodetrixStudio/CapacitorGoogleAuth/pull/345.

File: package.json (hunk 1 above)
Comment (KerimM) [reply package.json]: Done

**Comment by KerimM** [comment]
> I'll have to remember to return #345.

done

**Comment by reslear** [comment]
@KerimM thank you 🔥

**Comment by ManuelCostanzo** [comment]
Hey, sorry I have a question, why are you doing

`CAP_PLUGIN(GoogleAuth, "GoogleAuth")`

when that macro needs three params?

**Comment by reslear** [comment]
hi @ManuelCostanzo That's not right, can you create a revert PR?

**Comment by KerimM** [comment]
> Hey, sorry I have a question, why are you doing
> 
> `CAP_PLUGIN(GoogleAuth, "GoogleAuth")`
> 
> when that macro needs three params?

I follow the migration [guide](https://capacitorjs.com/docs/updating/plugins/6-0)

> Breaking changes in code
iOS
Remove removeAllListeners method
If you plugin had CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnPromise); in the .m file, it can be removed now, the method is now available for all plugins without defining it.

**Comment by reslear** [comment]
but need remove only core method `removeAllListeners` :) pls revert PR

**Comment by gasci** [comment]
Is there an update regarding this topic? 3.4.0-rc.1 fails @KerimM

**Comment by reslear** [comment]
hi @gasci check please v3.4.0-rc.2

**Comment by gasci** [comment]
I am getting this error for some reason
Screenshot 2024-04-29 at 11 44 23

**Comment by gasci** [comment]
For some reason rc.2 isn't available. Am I doing something wrong?
Screenshot 2024-04-29 at 11 56 27

**Comment by gasci** [comment]
And is it possible to align the major version with capacitor version. e.g. 6.0.0 @reslear ?

**Comment by mburger81** [comment]
When will we see this plugin as stable release for capacitor@6, thx you a lot for your work

**Comment by gasci** [comment]
I believe it is breaking on M1 computers and there a few major bugs that prevent this plugin to be stable for now. @mburger81 

I created issues as I detect them:
https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/363

**Comment by mburger81** [comment]
Ok but this issue has as source the @capacitor@6 update? @gasci if yes thats obviously bad, thx for your response

**Comment by gasci** [comment]
I haven't used this plugin before v6.

**Comment by mburger81** [comment]
@gasci it seems you have fixed the problem here
https://github.com/CodetrixStudio/CapacitorGoogleAuth/issues/363#issuecomment-2193933781

so we should be able to have a stable release?

**Comment by gasci** [comment]
I would say so. Yes, I fixed the issue.

**Comment by mburger81** [comment]
@gasci can we have a official release for capacitor@6? I mean at the end of august you can't upload to Gogole Play Store if you don't use Android14+ which means for use to upgrade to capacitor@6.

**Comment by gasci** [comment]
Hi, I don't think I have the permissions to do that. Please reach out to the original maintainers.

**Comment by mburger81** [comment]
Oh yea thats true, sorry @gasci 


@KerimM @reslear can you please do a official release without the `-rc.x `?

**Comment by reslear** [comment]
@mburger81 you have to test it first

**Comment by gasci** [comment]
I tested this version and I am running it without any problems

**Pull request merged**